architecture/adr-list/adr-002-identity-transactions.md

```diff
@@ -0,0 +1,274 @@
+
+- **DID**: the hardcoded string `did:` to indicate the identifier is a DID
+- **`cheqd` DID method**: the hardcoded string `cheqd:` indicating that the identifier uses the `cheqd` DID method specification.
+- **Namespace**: A string that identifies the name of the primary cheqd ledger ("mainnet"), followed by a `:`. The namespace string may optionally have a secondary ledger name prefixed by a `:` following the primary name. If there is no secondary ledger element, the DID resides on the primary ledger ("mainnet"), else it resides on the secondary ledger. By convention, the primary is a production ledger while the secondary ledgers are non-production ledgers (e.g. testnet) associated with the primary ledger.
+- **Namespace Identifier**: A self-certified identifier unique to the given cheqd DID ledger namespace.
+
+The components are assembled as follows:
+
+`did:cheqd:<namespace>:<namespace identifier>`
+
+Some examples of `did:cheqd` method identifiers are:
+
+- A DID written to the cheqd mainnet ledger: `did:cheqd:mainnet:7Tqg6BwSSWapxgUDm9KKgg`
+- A DID written to the cheqd testnet ledger: `did:cheqd:testnet:6cgbu8ZPoWTnR5Rv5JcSMB`
+
+### General structure of transaction requests
+
+All identity requests will have the following format:
+
+```json
+{
+    "data": { "<request data for writing a transaction to the ledger>" },
+    "authors": ["<identifier>", "..."],
+    "signatures": [
+        "<verification method id>": "<signature>",
+      ],
+    "metadata": {}
+}
+```
+
+- **`data`**: Data requested to be written to the ledger, specific for each request type.
+- **`authors`**: Authors identifiers (DID) list for this entity. There should be a new DIDs or an existing DIDs, for existing entities.
+- **`signatures`**: `data` should be signed by all `authors` private key. This field contains a dict there authors key's id from `DIDDoc.authentication` is a key, and the signature is a value.
+- **`metadata`**: Dictionary with additional metadata fields. Empty for now. This fields provides extensibility in the future, e.g., it can contain `protocolVersion` or other relevant metadata associated with a request.
+
+## List of transactions and details
+
+### `DID` transactions
+
+[Decentralized Identifiers \(DIDs\) are a W3C specification](https://www.w3.org/TR/did-core/) for identifiers that enable verifiable, decentralized digital identity.
+
+DIDDoc format conforms to [DIDDoc spec](https://www.w3.org/TR/did-core/#representations).
+The request can be used for creation of new DIDDoc, setting, and rotation of verification key.
+
+#### DIDDoc
+
+1. **`id`**: Target DID as base58-encoded string for 16 or 32 byte DID value. In the ledger, we store only an identifier without specifying the method and namespace.
+2. **`controller`** (optional): A list of base58-encoded identifier strings.
+3. **`verificationMethod`** (optional): A list of Verification Methods
+4. **`authentication`** (optional): A list of Verification Methods or strings with key aliases
```

Comment:
Do we want to use here just list of string as it's implemented in cosmos-cash?
https://github.com/allinbits/cosmos-cash/blob/e8ffcc38d76111a0fda29266f36e364b645531eb/proto/did/did.proto#L44

FYI, @ankurdotb

Reply:
I completely agree with this idea. In this way, we trim the original DiDDoc, but we keep all business cases, as well as make this interface more intuitive.